kernel/exit.c

```diff
@@ -243,7 +246,7 @@ static bool reap_if_needed(struct task *task, struct siginfo_ *info_out, struct
 int do_wait(int idtype, pid_t_ id, struct siginfo_ *info, struct rusage_ *rusage, int options) {
     if (idtype != P_ALL_ && idtype != P_PID_ && idtype != P_PGID_)
         return _EINVAL;
-    if (options & ~(WNOHANG_|WUNTRACED_|WEXITED_|WCONTINUED_|WNOWAIT_))
+    if (options & ~(WNOHANG_|WUNTRACED_|WEXITED_|WCONTINUED_|WNOWAIT_|__WALL_))
```

Comment:
Should there be something checking for and handling `__WALL_`, or is it OK to ignore?

Reply:
I think in recent kernel versions this is implied or something like that, but GDB passes it in anyways to be safe so we just silently accept it as a valid flag.